src/Futhark/Actions.hs

```diff
@@ -255,11 +329,46 @@ compileMulticoreAction fcfg mode outpath =
           liftIO $ writeFile hpath $ cPrependHeader header
           liftIO $ writeFile cpath $ cPrependHeader impl
         ToExecutable -> do
-          liftIO $ writeFile cpath $ cPrependHeader $ MulticoreC.asExecutable cprog
-          runCC cpath outpath ["-O", "-std=c99"] ["-lm", "-pthread"]
+          liftIO $ writeFile cpath $ MulticoreC.asExecutable cprog
+          runCC cpath outpath ["-O3", "-std=c99"] ["-lm", "-pthread"]
```

Comment:
Remove these changes.

Reply:
Is it just the O3 optimisation I am meant to be taking out?

Reply:
Also the removal of `cPrependHeader`.